marsha/core/models.py

```diff
@@ -0,0 +1,392 @@
+            "if closed captioning (for death or hard of hearing viewers) "
+            "is on for this subtitle track"
+        ),
+    )
+
+    # annotate inherited fields for mypy
+    video: Video
+    language: str
+
+    class Meta:
+        """Options for the ``SubtitleTrack`` model."""
+
+        verbose_name: str = _("subtitles track")
+        verbose_name_plural: str = _("subtitles tracks")
+        unique_together: UniqueTogether = [
+            ("video", "language", "has_closed_captioning")
+        ]
+
+
+class SignTrack(VideoTrackBase):
+    """Model representing a signs language track for a video."""
+
+    # annotate inherited fields for mypy
+    video: Video
+    language: str
+
+    class Meta:
+        """Options for the ``SignTrack`` model."""
+
+        verbose_name: str = _("signs language track")
+        verbose_name_plural: str = _("signs language tracks")
+        unique_together: UniqueTogether = [("video", "language")]
+
+
+class Playlist(BaseModel):
+    """Model representing a playlist which is a list of videos."""
+
+    organization: Organization = models.ForeignKey(
+        to=Organization,
+        related_name="playlists",
+        # don't delete a playlist if its organization is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    author: User = models.ForeignKey(
+        to=User,
+        related_name="created_playlists",
+        # don't delete a playlist if its author is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    editors: M2MType[User] = models.ManyToManyField(
+        to=User,
+        through="PlaylistRight",
+        related_name="managed_playlists",
+        verbose_name=_("editors"),
+        help_text=_("users allowed to manage this playlist"),
+    )
+    is_public: bool = models.BooleanField(
+        default=False,
+        verbose_name=_("is public"),
+        help_text=_("if this playlist can be viewed without any access control"),
+    )
+    duplicated_from: "Playlist" = models.ForeignKey(
+        to="self",
+        related_name="duplicates",
+        verbose_name=_("duplicate from"),
+        help_text=_("original playlist this one was duplicated from"),
+        # don't delete a playlist if the one it was duplicated from is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    videos: M2MType[Video] = models.ManyToManyField(
+        to=Video,
+        through="PlaylistVideo",
+        related_name="playlists",
+        verbose_name=_("videos"),
+        help_text=_("videos in this playlist"),
+    )
+
+    # `related_name` of fields pointing to this model, for typing
+    duplicates: ReverseFKType["Playlist"]
+    videos_links: ReverseFKType["PlaylistVideo"]
+    users_rights: ReverseFKType["PlaylistRight"]
+
+    class Meta:
+        """Options for the ``Playlist`` model."""
+
+        verbose_name: str = _("playlist")
+        verbose_name_plural: str = _("playlists")
+
+
+class PlaylistVideo(BaseModel):
```

Comment:
Why not just Playlist? 
If you want to add video it should be VideoPlaylist in english?

Reply:
It represents a video a in a playlist. Not the video or the playlist itself. And then used as the through table between these two models.

So for me, `VideoPlaylist` or `PlaylistVideo` are the same.